homeassistant/components/resolution_center/issue_handler.py

```diff
@@ -0,0 +1,86 @@
+"""The resolution center integration."""
+from __future__ import annotations
+
+import dataclasses
+
+from awesomeversion import AwesomeVersion, AwesomeVersionStrategy
+
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DOMAIN
+from .issue_registry import async_get as async_get_issue_registry
+from .models import Issue, IssueSeverity
+
+
+@callback
+def async_create_issue(
+    hass: HomeAssistant,
+    domain: str,
+    issue_id: str,
+    *,
+    breaks_in_ha_version: str | None = None,
+    is_fixable: bool,
+    learn_more_url: str | None = None,
+    severity: IssueSeverity,
+    translation_key: str,
+    translation_placeholders: dict[str, str] | None,
+) -> None:
+    """Create an issue, or replace an existing one."""
+    # Verify the breaks_in_ha_version is a valid version string
+    if breaks_in_ha_version:
+        AwesomeVersion(
+            breaks_in_ha_version,
+            ensure_strategy=AwesomeVersionStrategy.CALVER,
+            find_first_match=False,
+        )
+
+    issue_registry = async_get_issue_registry(hass)
+
+    issue_entry = issue_registry.async_get_or_create(domain, issue_id)
+
+    issue = Issue(
+        breaks_in_ha_version=breaks_in_ha_version,
+        domain=domain,
+        issue_id=issue_id,
+        dismissed=issue_entry.is_dismissed,
+        dismissed_version=issue_entry.dismissed_version,
+        is_fixable=is_fixable,
+        learn_more_url=learn_more_url,
+        severity=severity,
+        translation_key=translation_key,
+        translation_placeholders=translation_placeholders,
+    )
+    hass.data[DOMAIN]["issues"][(domain, issue_id)] = issue
```

Comment:
We should only maintain these objects in the `issue_registry` and not duplicate it in `hass.data[DOMAIN]`

Reply:
If you want to use it only to store dismissable, you should call it a store, not a registry.

Reply:
ff84887